### DISCUSSION TITLE
Response code 200, no changes being made

### DISCUSSION THREAD

**Discussion by alexhoopes** [netbox-community/netbox#8898] [Q&A] [answered]
I am using requests from Python to run my API requests.
Trying to update a device using https://netbox.domain.com/api/dcim/devices/{id}
I get http response 200, but no changes are actually being made. I am trying to push things such as {"serial": "1234"}

API request looks like this:
```py
editpayload = {'serial': '1234'}
def get_edit(hostname, editpayload, headers, verbose, nbid):
    edit = requests.patch(
        url = f'https://url.com/api/dcim/devices/{nbid}',
        headers=headers, timeout=60, verify=False, data=json.dumps(editpayload))
    editstatus = edit.status_code

request = get_edit(hostname, editpayload, headers, verbose, nbid)
```
Any ideas what I could be doing wrong here? Not sure why I get a response code of 200, but nothing on Netbox has actually changed. The request is there in the device log, but no change to the serial has happened.

**Comment by markkuleinio** [comment, marked as answer]
Does it change if you add the missing ending slash in the URL?

Also, to ensure correct content-type headers, you can use `json=editpayload` instead of converting to JSON yourself.

**Reply by alexhoopes** [reply]
It does. Thank you much!
